Firmware/ultralcd.cpp

```diff
@@ -902,9 +902,24 @@ void lcd_commands()
         switch (lcd_commands_step)
         {
         case 0:
-            lcd_commands_step = 3;
+            lcd_commands_step = 6;
             [[fallthrough]];
 
+        case 6:
+            enquecommand_P(G28W);
+            lcd_commands_step = 5;
+            break;
+
+        case 5:
+            enquecommand_P(PSTR("G1 X125 Y105 Z1 F8000"));
+            lcd_commands_step = 4;
+            break;
+
+        case 4:
+            st_synchronize();
```

Comment:
This looks wrong. st_synchronise will block before the previos gcode commands get a chance to be parsed and added to the planner properly. I would replace this with an M400 call.

Reply:
Actually, nevermind. The st_sync will only execute once the cmdqueue is empty, so it should be safe. However, I still think the other optimization I mentioned should be applied at some point in the future, if not now.

Reply:
Right the M400 cost 10 bytes.

Reply:
I would replace this with

```
if(!blocks_queued()) { lcd_commands_step = 3; }
break;
```

so this avoids the nested wait loop. We already check for the cmdbuffer at the uppermost if(), so it we should be covered.

But it's a minor thing. This should also be fine.